Flaky-test: PulsarFunctionsJavaThreadTest.testJavaLoggingFunction

<!--- 

Instructions for reporting a flaky test using this issue template:

1. Replace test_class in title and body with the short test class name WITHOUT the package name.
2. Replace test_method in title and body with the test method that failed. Multiple methods are flaky, remove the content that refers to the test method.
3. Replace "url here" with a url to an example failure. In the Github Actions workflow run logs, you can right click on the line number to copy a link to the line. Example of such url is https://github.com/apache/pulsar/pull/8892/checks?check_run_id=1531075794#step:9:377 . The logs are available for a limited amount of time (usually for a few weeks).
4. Replace "relevant parts of the exception stacktrace here" with the a few lines of the stack trace that shows at least the exception message and the line of test code where the stacktrace occurred.
5. Replace "full exception stacktrace here" with the full exception stacktrace from logs. This section will be hidden by default.
6. Remove all unused fields / content to unclutter the reported issue. Remove this comment too.

-->
 PulsarFunctionsJavaThreadTest.testJavaLoggingFunction is flaky. It fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/6225857381)

```
2022-04-29T10:26:05.4063041Z [ERROR] testJavaLoggingFunction(org.apache.pulsar.tests.integration.functions.java.PulsarFunctionsJavaThreadTest)  Time elapsed: 50.242 s  <<< FAILURE!
2022-04-29T10:26:05.4064216Z java.lang.NullPointerException: Cannot invoke "org.apache.pulsar.client.api.Message.getValue()" because "msg" is null
2022-04-29T10:26:05.4131524Z 	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.publishAndConsumeMessages(PulsarFunctionsTest.java:1583)
2022-04-29T10:26:05.4132844Z 	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.testLoggingFunction(PulsarFunctionsTest.java:1506)
2022-04-29T10:26:05.4133941Z 	at org.apache.pulsar.tests.integration.functions.java.PulsarFunctionsJavaTest.testJavaLoggingFunction(PulsarFunctionsJavaTest.java:111)
2022-04-29T10:26:05.4135136Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-04-29T10:26:05.4135817Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-04-29T10:26:05.4136564Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-29T10:26:05.4137167Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-04-29T10:26:05.4137752Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
2022-04-29T10:26:05.4138406Z 	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
2022-04-29T10:26:05.4139003Z 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
2022-04-29T10:26:05.4139577Z 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
2022-04-29T10:26:05.4140122Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-04-29T10:26:05.4140702Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-04-29T10:26:05.4141652Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-04-29T10:26:05.4142141Z 	at java.base/java.lang.Thread.run(Thread.java:833)
```

this remains very flaky
```
  Error:  Tests run: 59, Failures: 1, Errors: 0, Skipped: 45, Time elapsed: 1,035.55 s <<< FAILURE! - in TestSuite
  Error:  testJavaLoggingFunction(org.apache.pulsar.tests.integration.functions.java.PulsarFunctionsJavaThreadTest)  Time elapsed: 178.075 s  <<< FAILURE!
  java.lang.AssertionError: expected [10] but found [0]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:797)
  	at org.testng.Assert.assertEquals(Assert.java:807)
  	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.getFunctionStats(PulsarFunctionsTest.java:1013)
  	at org.apache.pulsar.tests.integration.functions.PulsarFunctionsTest.testLoggingFunction(PulsarFunctionsTest.java:1521)
  	at org.apache.pulsar.tests.integration.functions.java.PulsarFunctionsJavaTest.testJavaLoggingFunction(PulsarFunctionsJavaTest.java:111)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```
at https://github.com/apache/pulsar/runs/8066475125?check_suite_focus=true#step:11:19252

--- a/bin/pulsar
+++ b/bin/pulsar
@@ -316,11 +316,8 @@ OPTS="$OPTS -Dpulsar.log.dir=$PULSAR_LOG_DIR"
 OPTS="$OPTS -Dpulsar.log.level=$PULSAR_LOG_LEVEL"
 OPTS="$OPTS -Dpulsar.log.root.level=$PULSAR_LOG_ROOT_LEVEL"
 OPTS="$OPTS -Dpulsar.routing.appender.default=$PULSAR_ROUTING_APPENDER_DEFAULT"
-
-
 # Configure log4j2 to disable servlet webapp detection so that Garbage free logging can be used
-PULSAR_LOG4J_CONF=${PULSAR_LOG4J_CONF:-"-Dlog4j2.is.webapp=false  -Dlog4j2.contextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector -Dlog4j2.enableThreadlocals=true -Dlog4j2.enableDirectEncoders=true"}
-OPTS="$OPTS $PULSAR_LOG4J_CONF"
+OPTS="$OPTS -Dlog4j2.is.webapp=false"
 
 # Functions related logging
 OPTS="$OPTS -Dpulsar.functions.process.container.log.dir=$PULSAR_LOG_DIR"

--- a/conf/log4j2.yaml
+++ b/conf/log4j2.yaml
@@ -53,7 +53,6 @@ Configuration:
     Console:
       name: Console
       target: SYSTEM_OUT
-      direct: true
       PatternLayout:
         Pattern: "%d{ISO8601_OFFSET_DATE_TIME_HHMM} [%t] %-5level %logger{36} - %msg%n"
 

--- a/distribution/server/pom.xml
+++ b/distribution/server/pom.xml
@@ -65,11 +65,6 @@
     </dependency>
 
     <dependency>
-      <groupId>com.lmax</groupId>
-      <artifactId>disruptor</artifactId>
-    </dependency>
-
-    <dependency>
       <groupId>org.apache.zookeeper</groupId>
       <artifactId>zookeeper-prometheus-metrics</artifactId>
       <version>${zookeeper.version}</version>

--- a/distribution/server/src/assemble/LICENSE.bin.txt
+++ b/distribution/server/src/assemble/LICENSE.bin.txt
@@ -543,8 +543,6 @@ The Apache Software License, Version 2.0
     - io.etcd-jetcd-core-0.5.11.jar
   * IPAddress
     - com.github.seancfoley-ipaddress-5.3.3.jar
-  * LMAX Disruptor
-    - com.lmax-disruptor-3.4.3.jar
   * RxJava
     - io.reactivex.rxjava3-rxjava-3.0.1.jar
   * RabbitMQ Java Client

